modules/vcf2maf/1.2/vcf2maf.smk

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Bruno Grande
+# Module Author:    Helena Winata
+# Contributors:     Ryan Morin
+
+
+##### SETUP #####
+
+import sys, os
+from os.path import join
+
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CONFIG`
+CFG = op.setup_module(
+    name = "vcf2maf",
+    version = "1.2",
+    subdirectories = ["inputs","decompressed","vcf2maf","crossmap","outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _vcf2maf_input_vcf,
+    _vcf2maf_decompress_vcf,
+    _vcf2maf_output_maf,
+    _vcf2maf_crossmap,
+    _vcf2maf_all
+
+VERSION_MAP = {
+    "grch37": "GRCh37",
+    "hg38": "GRCh38",
+    "hs37d5": "GRCh37"
+}
+
+##### RULES #####
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _vcf2maf_input_vcf:
+    input:
+        vcf_gz = CFG["inputs"]["sample_vcf_gz"]
+    output:
+        vcf_gz = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}.vcf.gz"
+    run:
+        op.relative_symlink(input.vcf_gz, output.vcf_gz)
+
+rule _vcf2maf_decompress_vcf:
+    input:
+        vcf_gz = str(rules._vcf2maf_input_vcf.output.vcf_gz)
+    output:
+        vcf = temp(CFG["dirs"]["decompressed"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}.vcf")
+    shell:
+        "gzip -dc {input.vcf_gz} > {output.vcf}" #this should work on both gzip and bcftools compressed files 
+
+rule _vcf2maf_run:
+    input:
+        vcf = CFG["dirs"]["decompressed"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}.vcf",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        vep_cache = CFG["inputs"]["vep_cache"]
+    output:
+        maf = CFG["dirs"]["vcf2maf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}.maf", 
+        vep = temp(CFG["dirs"]["decompressed"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}.vep.vcf")
+    log:
+        stdout = CFG["logs"]["vcf2maf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}_vcf2maf.stdout.log",
+        stderr = CFG["logs"]["vcf2maf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{base_name}_vcf2maf.stderr.log",
+    params:
+        opts = CFG["options"]["vcf2maf"],
+        build = lambda w: VERSION_MAP[w.genome_build],
+        custom_enst = op.switch_on_wildcard("genome_build", CFG["switches"]["custom_enst"])
```

Comment:
Does this work with an empty file for custom_enst ?

Reply:
Yes, it does. When I removed ENST entry for demo sample to address your previous comment, I run the whole module again and confirmed it worked with an empty file. I did not test on GAMBL not to create there an output file that will be misleading.